### PR TITLE
Fence Collider/Outline Fix

### PR DESCRIPTION
**Pull request by matthewperiut**
I made it so that the collision/outline shape isn't so large for fences, and I thought it would be a good addition to AnnoyanceFix
Video demonstration with old and new

https://github.com/telvarost/AnnoyanceFix-StationAPI/assets/18296791/c43550f7-248b-49f7-92e6-85590e728db3

### CODE REVIEW

**Comment by matthewperiut** [comment]
coincidentally did the same thing as Vanilla Block Enhancements ;-; realizing

**Comment by 2zqa** [comment]
@matthewperiut That's no problem! AnnoyanceFix differs from Vanilla Block Enhancement in the way that we don't use custom blocks; so if you accidentally load your world without AnnoyanceFix, at least all your blocks are still there :) we'd love to integrate your PR if you're still interested. I see you deleted the branch, but luckily it seems the diff is still available in the changes tab here.

File: src/main/java/com/github/telvarost/annoyancefix/mixin/FenceMixin.java [hunk 1]
```diff
@@ -26,4 +28,64 @@ private void annoyanceFix_canPlaceAt(Level arg, int i, int j, int k, CallbackInf
...
+        if (collider) {
+            box.maxY += 0.5F;
+        }
+
+        return box;
+    }
+
+    @Inject(method = "getCollisionShape", at = @At("RETURN"), cancellable = true)
+    public void annoyanceFix_getCollisionShape(Level level, int x, int y, int z, CallbackInfoReturnable<Box> cir) {
+        if (Config.ConfigFields.fenceFixesEnabled) {
```
Comment (2zqa) [inline src/main/java/com/github/telvarost/annoyancefix/mixin/FenceMixin.java]: Could you add a separate setting for the collider/outline fix? E.g. "fenceBoundingBoxFixesEnabled"

**Review by 2zqa — APPROVED** [review]
Thank you for your contribution! We'll be incorporating it in the next version, which is likely 1.13.1.

File: src/main/java/com/github/telvarost/annoyancefix/mixin/FenceMixin.java (hunk 1 above)
Comment (matthewperiut) [reply src/main/java/com/github/telvarost/annoyancefix/mixin/FenceMixin.java]: yes

**Comment by 2zqa** [comment]
Just tested it, works like a charm! 👏 Great addition. For a future version we can try to implement even better hitbox detection for corner pieces, but this will definitely be added the way it is now.

https://github.com/telvarost/AnnoyanceFix-StationAPI/assets/25235249/0098d69a-cadc-492b-a561-e0979ee4c1fe

**Comment by matthewperiut** [comment]
so, yes we could implement a better hitbox detection method, but that would require using multiple bounding boxes, and that's something that was added much later to the game, it would require a pretty large overhaul of the game

**Comment by 2zqa** [comment]
That would probably be overkill then. We can leave it at this for now

**Pull request merged**